trigger/contrib/commando/plugins/gather_info.py

```diff
@@ -0,0 +1,143 @@
+"""
+Plugin for gathering info from devices. Only Cisco currently (including ASA).
+
+Will try to parse as much info from commands and place as key-values under
+host results. Currently only being done via ``show version`` output. Will
+also run all commands listed under ``to_cisco()`` depending on the platform
+and can be accessed via each host's ``commands_ran`` dictionary under results.
+
+Use-case is likely limited for well-established networks, but for
+environments (MSP) where one might be handed list of IP addresses with
+nothing more than device type and credentials this can provide extremely
+helpful to just build a quick NetDevices CSV and gather a bunch of info
+quickly. For example::
+
+    >>> import os
+    >>> from trigger import tacacsrc
+    >>> from trigger.conf import settings
+    >>> from trigger.netdevices import NetDevices
+    >>> from trigger.contrib.commando.plugins import gather_info
+    >>>
+    >>>
+    >>> settings.NETDEVICES_SOURCE = os.path.abspath('netdevices.csv')
+    >>> settings.DEFAULT_REALM = 'MyRealm'
+    >>> os.environ['TRIGGER_ENABLEPW'] = \
+    >>>     tacacsrc.get_device_password(settings.DEFAULT_REALM).password
+    >>>
+    >>> gi = GatherInfo(devices=device_list)
+    >>> gi.run()
+    >>>
+    >>> for host, value in gi.results:
+    >>>     print("[{}] {}: {}".format(host,
+                                       value['hostname'],
+                                       value['serial_no']))
+
+GatherInfo.results will look something like::
+
+    {"host1": {
+                "hostname": "host1",
+                "serial_no": "xxxyyy",
+                # ..other parsed attrs..
+                "commands_ran": {
+                                  "show version": "[OUTPUT]",
+                                  "show run": "[OUTPUT]",
+                                  # ..and so on..
+                                }
+              }
+    }
+
+
+"""
+
+import re
+from trigger.cmds import Commando
+from twisted.python import log
+
+
+task_name = 'gather_info'
+
+
+def xmlrpc_gather_info(*args, **kwargs):
+    """Gather info on specified devices"""
+    log.msg('Creating GatherInfo')
+    gi = GatherInfo(*args, **kwargs)
+    d = gi.run()
+    return d
+
+
+class GatherInfo(Commando):
```

Comment:
Since this is currently Cisco-only, can you please add `vendors = ['cisco']` to this class? Then this will be good to go! Thanks!